### PR TITLE
Cancel immediately and doc comments

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/InternalClient.cs [hunk 1]
```diff
@@ -176,6 +176,8 @@ public void SetConnectionStatusChangeHandler(Action<ConnectionInfo> statusChange
         {
             if (Logging.IsEnabled)
                 Logging.Enter(this, callback, userContext, nameof(SetDesiredPropertyUpdateCallbackAsync));
+            
+            cancellationToken.ThrowIfCancellationRequested();
```
Comment (drwill-ms) [inline iothub/device/src/InternalClient.cs:180]: In many cases we weren't checking for cancellation immediately.

File: iothub/device/src/IotHubDeviceClient.cs [hunk 2]
```diff
@@ -103,8 +103,8 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         /// Sets a new delegate for the connection status changed callback. If a delegate is already associated,
         /// it will be replaced with the new delegate. Note that this callback will never be called if the client is configured to use
         /// HTTP, as that protocol is stateless.
-        /// <param name="statusChangeHandler">The name of the method to associate with the delegate.</param>
```
Comment (drwill-ms) [inline iothub/device/src/IotHubDeviceClient.cs:106]: We had cases of param definitions inside the summary

File: iothub/device/src/IotHubDeviceClient.cs [hunk 3]
```diff
@@ -116,35 +116,34 @@ public void SetConnectionStatusChangeHandler(Action<ConnectionInfo> statusChange
...
         /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
-        /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
         /// </summary>
+        /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
         public Task OpenAsync(CancellationToken cancellationToken = default) => InternalClient.OpenAsync(cancellationToken);
 
         /// <summary>
         /// Sends an event to a hub
         /// </summary>
+        /// <remarks>
```
Comment (drwill-ms) [inline iothub/device/src/IotHubDeviceClient.cs:126]: I try to follow the order of:
- summary
- remarks
- params
- returns
- exceptions
- example

File: iothub/device/src/IotHubDeviceClient.cs [hunk 4]
```diff
@@ -348,17 +354,19 @@ public Task UpdateReportedPropertiesAsync(TwinCollection reportedProperties, Can
...
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The file upload details to be used with the Azure Storage SDK in order to upload a file from this device.</returns>
+        /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
         public Task<FileUploadSasUriResponse> GetFileUploadSasUriAsync(
             FileUploadSasUriRequest request,
             CancellationToken cancellationToken = default)
             => InternalClient.GetFileUploadSasUriAsync(request, cancellationToken);
 
         /// <summary>
-        /// Notify IoT hub that a device's file upload has finished. See
```
Comment (drwill-ms) [inline iothub/device/src/IotHubDeviceClient.cs:357]: Moved to seealso

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/device/src/IotHubDeviceClient.cs (hunk 3 above)
Comment (andyk-ms) [reply iothub/device/src/IotHubDeviceClient.cs:126]: Good order to follow.

**Review by andyk-ms — APPROVED** [review]

**Pull request merged**